client/app/templates/templates.service.js

```diff
@@ -73,4 +74,17 @@ export function TemplatesServiceFactory(CollectionsApi) {
 
     return queryFilters;
   }
+  function createTemplate(template) {
+    return new Promise(function (resolve, reject) {
```

Comment:
This can be really simplified by using the fact that CollectionsApi methods already return a promise, since nothing special is happening in the callbacks

```js
return CollectionsApi.post(collection, null, {}, template);
```

Does essentially the same thing.